specs/notifications-api/notifications-api.md

```diff
@@ -0,0 +1,328 @@
+Aside from the first and third bullet, this is identical to the UWP experience today. They simply use our Interop class to create the notifier, which for Win32 automatically registers their app and calls the correct CreateToastNotifier with their aumid.
+
+To receive activation, we haven't been able to get a 100% converged experience...
+
+* UWP apps: They would still use App.xaml.cs OnActivated as they do today
+* Win32 MSIX/sparse apps: They would first have to add a COM server registration in their app manifest (maybe tooling can eliminate this somehow), and then they subscribe to ToastNotificationManagerInterop.OnActivated
+* Win32 apps: They subscribe to ToastNotificationManagerInterop.OnActivated
+
+# Examples
+
+## Sending toast API experience
+
+First, developers would install the Reunion NuGet package.
+
+Then, developers construct a notification as usual, but call the Interop version of ToastNotificationManager to create the toast notifier. With that one change, toasts will work on Win32 without doing anything additional!
+
+```csharp
+// Construct the content (same as today)
+ToastContent content = new ToastContentBuilder()
+    .AddToastActivationInfo("picOfHappyCanyon", ToastActivationType.Foreground)
+    .AddText("Andrew sent you a picture")
+    .AddText("Check this out, Happy Canyon in Utah!")
+    .GetToastContent();
+
+// Create the notification (same as today)
+ToastNotification notification = new ToastNotification(content.GetXml());
+
+// And send it (Interop is the only difference)
+ToastNotificationManagerInterop.CreateToastNotifier().Show(notification);
+```
+
+## Receiving activation API experience
+
+### UWP apps
+
+UWP apps would receive activation as they do today, within their **App.xaml.cs** OnActivated method.
+
+```csharp
+protected override void OnActivated(IActivatedEventArgs e)
+{
+    // Handle toast activation
+    if (e is ToastNotificationActivatedEventArgs toastActivationArgs)
+    {
+        // Obtain the arguments from the toast
+        string args = toastActivationArgs.Argument;
+
+        // Obtain any user input (text boxes, menu selections) from the toast
+        ValueSet userInput = toastActivationArgs.UserInput;
+ 
+        // TODO: Show the corresponding content
+    }
+}
+```
+
+### Win32 MSIX/Sparse apps
+
+First, in your Package.appxmanifest, add:
+
+1. Declaration for xmlns:com
+1. Declaration for xmlns:desktop
+1. In the IgnorableNamespaces attribute, com and desktop
+1. com:Extension for the COM activator using a new GUID of your choice. Be sure to include the Arguments="-ToastActivated" so that you know your launch was from a toast
+1. desktop:Extension for windows.toastNotificationActivation to declare your toast activator CLSID (the GUID from #4 above).
+
+Then, **in your app's startup code** (WinMain), subscribe to the **OnActivated** event.
+
+```csharp
+
+int main(int argc, char* argv[], char* envp[])
```

Comment:
Is this code C# or C++? That isn't what a main function in C# looks like, and WinMain is used in C++ apps only. If it's actually C#, it should be `Main(string[] args)`. Furthermore,  the `init_apartment()` call is C++-specific; to specify an STA apartment, put `[STAThread]` attribute on the Main method.

Reply:
Ah this is c++ sample code. Not c#. Will change to update that.

Reply:
Good catch. I'll move this to c# syntax